packages/ipfs-http-client/src/add/index.js

```diff
@@ -1,51 +1,36 @@
 'use strict'
 
-const ndjson = require('iterable-ndjson')
 const CID = require('cids')
-const configure = require('../lib/configure')
-const toIterable = require('stream-to-it/source')
+const merge = require('merge-options')
 const { toFormData } = require('./form-data')
 const toCamel = require('../lib/object-to-camel')
+const configure = require('../lib/configure')
 
-module.exports = configure(({ ky }) => {
-  return async function * add (input, options) {
-    options = options || {}
-
-    const searchParams = new URLSearchParams(options.searchParams)
-
-    searchParams.set('stream-channels', true)
-    if (options.chunker) searchParams.set('chunker', options.chunker)
-    if (options.cidVersion) searchParams.set('cid-version', options.cidVersion)
-    if (options.cidBase) searchParams.set('cid-base', options.cidBase)
-    if (options.enableShardingExperiment != null) searchParams.set('enable-sharding-experiment', options.enableShardingExperiment)
-    if (options.hashAlg) searchParams.set('hash', options.hashAlg)
-    if (options.onlyHash != null) searchParams.set('only-hash', options.onlyHash)
-    if (options.pin != null) searchParams.set('pin', options.pin)
-    if (options.progress) searchParams.set('progress', true)
-    if (options.quiet != null) searchParams.set('quiet', options.quiet)
-    if (options.quieter != null) searchParams.set('quieter', options.quieter)
-    if (options.rawLeaves != null) searchParams.set('raw-leaves', options.rawLeaves)
-    if (options.shardSplitThreshold) searchParams.set('shard-split-threshold', options.shardSplitThreshold)
-    if (options.silent) searchParams.set('silent', options.silent)
-    if (options.trickle != null) searchParams.set('trickle', options.trickle)
-    if (options.wrapWithDirectory != null) searchParams.set('wrap-with-directory', options.wrapWithDirectory)
-    if (options.preload != null) searchParams.set('preload', options.preload)
-    if (options.fileImportConcurrency != null) searchParams.set('file-import-concurrency', options.fileImportConcurrency)
-    if (options.blockWriteConcurrency != null) searchParams.set('block-write-concurrency', options.blockWriteConcurrency)
+module.exports = configure((api) => {
+  return async function * add (input, options = {}) {
+    const progressFn = options.progress
+    options = merge(
+      options,
+      {
+        'stream-channels': true,
+        progress: Boolean(progressFn),
+        hash: options.hashAlg // TODO fix this either is hash or hashAlg
+      }
+    )
 
-    const res = await ky.post('add', {
+    const res = await api.ndjson('add', {
```

Comment:
This is something to tidy up in a subsequent PR, but I think given the `fetch` pattern is:

```javascript
const res = await fetch('url')
const body = await res.json()
// ... do something with body
```

a more idiomatic approach here might be:

```javascript
const res = await api.post({...})

for await (const file of res.ndjson()) {
  //... do something with file
}
```

Reply:
I’d vote for having the res.ndjson put in a variable first in a separate line so that stack traces are easier to read

Reply:
So that would look like this:
```js
const res = await api.post({...})

const ndjsonRes = res.ndjson()

for await (const file of ndjsonRes) {
  //... do something with file
}
```

Reply:
@hugomrdias @achingbrain Curious for your thoughts on this when you get a chance. Thanks!

Reply:
i would prefer to keep this. ty
alex comment will be improved later.

Reply:
When you say keep this, do you mean your current approach as opposed to Alex's suggestion?

Or do you mean that in general you prefer the style where multiple operations happen in a single line?